control/control.go

```diff
@@ -395,16 +395,41 @@ func (p *pluginControl) SwapPlugins(in *core.RequestedPlugin, out core.Cataloged
 		return err
 	}
 
+	// Make sure plugin types and names are the same
+	if lp.TypeName() != out.TypeName() || lp.Name() != out.Name() {
+		fmt.Println(lp.TypeName())
+		fmt.Println(out.TypeName())
+		fmt.Println(lp.Name())
+		fmt.Println(out.Name())
+		serr := serror.New(errors.New("Plugin types and names must match."))
+		serr.SetFields(map[string]interface{}{
+			"in Type":  lp.TypeName(),
+			"out Type": out.TypeName(),
```

Comment:
I think by convention these keys shouldn't contain spaces.

Technically it's valid, but I'd prefer something like `in_type`.